### PR TITLE
Implementing and Documenting GEP 746 (TLS Updates)

### LINKED ISSUE #746: GEP: Replace Certificate Refs on HTTPRoute with Cross Namespace Refs from Gateway
**What would you like to be added**:
This proposes replacing Certificate Refs on HTTPRoute with the ability for Gateways to reference certificates in other namespaces. This is not a complete replacement, but it does significantly reduce the potential for confusion. 

**Why this is needed**:
This is a confusing part of the API that does not appear to be supported by any implementation yet. We can always choose to add this concept back if it's necessary, but leaving it in for v1alpha2 may leave us stuck with this feature for a long time.

This is a follow up on my [original proposal doc](https://docs.google.com/document/d/1Cv95XFCL6S_9pIyS0drnsDLsfinWc2tHOFl_x3-_SWI/edit).

### PR DESCRIPTION
**Pull request by robscott**
**What type of PR is this?**
/kind feature

**What this PR does / why we need it**:
This implements and documents GEP #746.

**Which issue(s) this PR fixes**:
Fixes #746

**Does this PR introduce a user-facing change?**:
```release-note
* Certificate refs from Gateways can now cross namespaces (with ReferencePolicy)
* Certificate refs no longer exist on HTTPRoute
```

Note: The CRD verification will fail on this PR until #754 gets in.

Deploy Preview: https://deploy-preview-768--kubernetes-sigs-gateway-api.netlify.app/v1alpha2/guides/tls/

/cc @hbagdi @howardjohn @jpeach @Miciah @youngnick

### CODE REVIEW

File: apis/v1alpha2/gateway_types.go [hunk 1]
```diff
@@ -316,8 +287,14 @@ type GatewayTLSConfig struct {
 
 	// CertificateRef is a reference to a Kubernetes object that contains a TLS
 	// certificate and private key. This certificate is used to establish a TLS
-	// handshake for requests that match the hostname of the associated listener.
-	// The referenced object MUST reside in the same namespace as Gateway.
+	// handshake for requests that match the hostname of the associated
+	// listener.
+	//
+	// When this references a resource in a different namespace than the
```
Comment (bowei) [inline apis/v1alpha2/gateway_types.go:293]: We should probably word this in the negative so it's clear:

CertificateRefs to a different namespace from the Gateway are denied UNLESS there
is a ReferencePolicy allowing the certificate to be attached. ...

Comment (robscott) [reply apis/v1alpha2/gateway_types.go]: Updated, thanks!

**Comment by robscott** [comment]
Thanks for the feedback @bowei! I think I've updated everything, PTAL.

File: examples/v1alpha2/tls-basic.yaml [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+  gatewayClassName: acme-lb
+  listeners:
+  - name: foo-https
+    protocol: HTTPS
+    port: 443
+    hostname: foo.example.com
+    tls:
+      certificateRef:
+        kind: Secret
+        group: ""
```
Comment (jpeach) [inline examples/v1alpha2/tls-basic.yaml:15]: consider omitting `group` since the default is "core"

**Review by jpeach — APPROVED** [review]
/lgtm

File: examples/v1alpha2/tls-basic.yaml (hunk 2 above)
Comment (robscott) [reply examples/v1alpha2/tls-basic.yaml:15]: I know we once had a default here, but it looks like we've removed it in favor of a generic ObjectRef, so I think this still needs to be specified.

**Pull request merged**